src/common/containers/repository-setup.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component, PropTypes } from 'react';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+import { createWebhook } from '../actions/webhook';
+import { Message } from '../components/forms';
+import Spinner from '../components/spinner';
+
+import styles from './container.css';
+
+class RepositorySetup extends Component {
+  componentDidMount() {
+    const { account, repo, isPending } = this.props;
+
+    if (!isPending) {
+      this.props.dispatch(createWebhook(account, repo));
+    }
+  }
+
+  render() {
+    const { fullName, isPending, success, error } = this.props;
+
+    if (success) {
+      this.props.router.push(`/${fullName}/builds`);
+    } else {
+      return (
+        <div className={styles.container}>
+          <Helmet
+            title={`Setting up ${fullName}`}
+          />
+          <h1>Setting up {fullName}</h1>
```

Comment:
This view is only visible for very short time (while webhook is being set up), so it maybe hard to even notice this title. But it's quite visible that it shows up and disappears again to be replaced by `/builds` view.

So I think I would remove the title and only keep loading spinner there, so the 'flash' of content is not so visible.